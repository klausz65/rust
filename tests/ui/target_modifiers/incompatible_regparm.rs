//@ aux-crate:wrong_regparm=wrong_regparm.rs
//@ compile-flags: --target i686-unknown-linux-gnu -Zregparm=1 -Cpanic=abort
//@ needs-llvm-components: x86
//@ revisions:error_generated allow_regparm_mismatch allow_any_mismatch allow_attr

//@[allow_regparm_mismatch] compile-flags: -Cunsafe-allow-abi-mismatch=regparm
//@[allow_any_mismatch] compile-flags: -Cunsafe-allow-abi-mismatch
//@[allow_regparm_mismatch] build-pass
//@[allow_any_mismatch] build-pass
//@[allow_attr] build-pass

#![crate_type = "lib"]
//[error_generated]~^ ERROR 12:1: 12:1: mixing `-Zregparm` will cause an ABI mismatch [incompatible_target_modifiers]
#![no_core]
#![feature(no_core, lang_items, repr_simd)]

#![cfg_attr(allow_attr, allow(incompatible_target_modifiers))]

fn foo() {
    wrong_regparm::somefun();
}
