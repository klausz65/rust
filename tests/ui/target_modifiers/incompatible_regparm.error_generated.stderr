error: mixing `-Zregparm` will cause an ABI mismatch
  --> $DIR/incompatible_regparm.rs:12:1
   |
LL | #![crate_type = "lib"]
   | ^
   |
   = help: `-Zregparm` modifies the ABI and Rust crates compiled with different values of this flag cannot be used together safely
   = note: `-Zregparm=1` in this crate is incompatible with `-Zregparm=2` in dependency `wrong_regparm`
   = note: alternatively, use `-Cunsafe-allow-abi-mismatch=regparm` to silence this error
   = note: `#[deny(incompatible_target_modifiers)]` on by default

error: aborting due to 1 previous error

